src/compression/b44/mod.rs

```diff
@@ -0,0 +1,964 @@
+        let channels = ChannelList::new(smallvec![channel.clone(), channel]);
+
+        let rectangle = IntegerBounds {
+            position: Vec2(0, 0),
+            size: Vec2(3, 2),
+        };
+
+        let (pixel_bytes, compressed, decompressed) =
+            test_roundtrip_noise_with(channels, rectangle);
+
+        assert_eq!(pixel_bytes.len(), 48);
+        assert_eq!(compressed.len(), 48);
+        assert_eq!(decompressed.len(), 48);
+        assert_eq!(pixel_bytes, decompressed);
+    }
+
+    #[test]
+    fn roundtrip_noise_mix_f32_f16_u32() {
+        let channels = ChannelList::new(smallvec![
+            ChannelDescription {
+                sample_type: SampleType::F32,
+                name: Default::default(),
+                quantize_linearly: false,
+                sampling: Vec2(1, 1),
+            },
+            ChannelDescription {
+                sample_type: SampleType::F16,
+                name: Default::default(),
+                quantize_linearly: false,
+                sampling: Vec2(1, 1),
+            },
+            ChannelDescription {
+                sample_type: SampleType::U32,
+                name: Default::default(),
+                quantize_linearly: false,
+                sampling: Vec2(1, 1),
+            }
+        ]);
+
+        let rectangle = IntegerBounds {
+            position: Vec2(-30, 100),
+            size: Vec2(322, 731),
+        };
+
+        let (pixel_bytes, compressed, decompressed) =
+            test_roundtrip_noise_with(channels, rectangle);
+
+        assert_eq!(pixel_bytes.len(), 2353820);
+        assert_eq!(compressed.len(), 2090578);
+        assert_eq!(decompressed.len(), 2353820);
+    }
+
+    #[test]
+    fn roundtrip_noise_mix_f32_f16_u32_tiny() {
+        let channels = ChannelList::new(smallvec![
+            ChannelDescription {
+                sample_type: SampleType::F32,
+                name: Default::default(),
+                quantize_linearly: false,
+                sampling: Vec2(1, 1),
+            },
+            ChannelDescription {
+                sample_type: SampleType::F16,
+                name: Default::default(),
+                quantize_linearly: false,
+                sampling: Vec2(1, 1),
+            },
+            ChannelDescription {
+                sample_type: SampleType::U32,
+                name: Default::default(),
+                quantize_linearly: false,
+                sampling: Vec2(1, 1),
+            }
+        ]);
+
+        let rectangle = IntegerBounds {
+            position: Vec2(0, 0),
+            size: Vec2(3, 2),
+        };
+
+        let (pixel_bytes, compressed, decompressed) =
+            test_roundtrip_noise_with(channels, rectangle);
+
+        assert_eq!(pixel_bytes.len(), 60);
+        assert_eq!(compressed.len(), 62);
+        assert_eq!(decompressed.len(), 60);
+    }
+
+    #[test]
+    fn border_on_multiview() {
+        // This test is hard to reproduce, so we use the direct image.
```

Comment:
Instead of the absolute path ` "/mnt/storage/code/repo/exrs/tests/images/valid/openexr/MultiView/Adjuster.exr"`, try using the relative path ` "tests/images/valid/openexr/MultiView/Adjuster.exr"`

Reply:
Facepalm… Too much enthusiasm from me.

Reply:
No worries :) I think we can merge now